.github/workflows/golangci-lint.yml

```diff
@@ -11,11 +11,10 @@ jobs:
     name: lint
     runs-on: ubuntu-latest
     steps:
+      - uses: actions/checkout@v4
       - name: Set up Go
         uses: actions/setup-go@v5
         with:
-          go-version: 1.20.x
-
-      - uses: actions/checkout@v4
```

Comment:
does this `uses: actions/checkout@v4` should be done in this commit?

Reply:
yes, this has to be moved, as `go.mod` will only exist after you checked out the file from git. After this commit, we do:
1) git clone
2) get go version from go.mod

The opposite order will cause a gh action failure as go.mod won't exist.